src/Adapter/MSTest.CoreAdapter/ObjectModel/TestMethod.cs

```diff
@@ -57,8 +57,20 @@ public TestMethod(MethodBase method, string name, string fullClassName, string a
 
             ManagedNameHelper.GetManagedName(method, out var managedType, out var managedMethod);
 
-            this.ManagedType = managedType;
-            this.ManagedMethod = managedMethod;
+            // ManagedNameHelpers currently does not support spaces in method names.
+            // If there are spaces in the method name, we'll use the legacy way to find the method.
+            if (!managedMethod.Contains(" "))
+            {
+                this.ManagedTypeName = managedType;
+                this.ManagedMethodName = managedMethod;
+            }
+        }
+
+        internal TestMethod(string managedTypeName, string managedMethodName, string name, string fullClassName, string assemblyName, bool isAsync)
```

Comment:
Should the constructor above be internal too? (only if it was added new)

Reply:
It's added on this preview.